module/tools/SystemConfiguration/src/SystemConfiguration.cpp

```diff
@@ -0,0 +1,356 @@
+
+                auto pos = groups.find(group);
+                if (pos == std::string::npos) {
+                    log<NUClear::INFO>(
+                        fmt::format("Group {} doesn't exist. Creating it and adding user {}", group, user));
+                    std::system(fmt::format("groupadd {}", group).c_str());
+                    std::system(fmt::format("useradd -m -G {} {}", group, user).c_str());
+                }
+                else {
+                    auto eol = groups.find("\n", pos);
+                    if (groups.substr(pos, eol - pos).find(user) == std::string::npos) {
+                        log<NUClear::INFO>(fmt::format("Adding user {} to group {}", user, group));
+                        std::system(fmt::format("useradd -m -G {} {}", group, user).c_str());
+                    }
+                }
+            }
+
+            /*******
+             * ZSH *
+             *******/
+            log<NUClear::INFO>("Ensuring zsh is configured");
+            fs::path home         = fs::path("/home") / user;
+            fs::path zprezto_root = home / ".zprezto";
+            if (!fs::exists(zprezto_root)) {
+                log<NUClear::INFO>(fmt::format("Cloning zprezto to {}", zprezto_root.string()));
+                std::system(fmt::format("git clone --recursive https://github.com/sorin-ionescu/prezto.git {}",
+                                        zprezto_root.string())
+                                .c_str());
+            }
+            else {
+                log<NUClear::INFO>(fmt::format("Pulling latest zprezto in {}", zprezto_root.string()));
+                std::system(fmt::format("git -C {} pull --recurse-submodules=yes", zprezto_root.string()).c_str());
+            }
+
+            log<NUClear::INFO>("Making zprezto symlinks");
+            for (const auto& p : fs::directory_iterator(zprezto_root / "runcoms")) {
+                if (p.is_regular_file() && p.path().filename().string()[0] == 'z') {
+                    fs::path target = p.path();
+                    fs::path link   = home / fmt::format(".{}", p.path().filename().string());
+
+                    log<NUClear::TRACE>("Making link {} -> {}", link.string(), target.string());
+                    fs::create_symlink(target, link);
+                }
+            }
+
+            log<NUClear::INFO>("Changing user shell to zsh");
+            std::system(fmt::format("chsh -s /usr/bin/zsh {}", user).c_str());
+
+            log<NUClear::INFO>("Appending fuzzy find scripts to zshrc");
+            std::ifstream ifs_zshrc(home / ".zshrc");
+            std::string line;
+            bool fuzzy_found = false;
+            while (std::getline(ifs_zshrc, line)) {
+                if (line.compare("# Source the fuzzy find scripts") == 0) {
+                    fuzzy_found = true;
+                    break;
+                }
+            }
+            ifs_zshrc.close();
+
+            if (fuzzy_found) {
```

Comment:
```suggestion
            if (!fuzzy_found) {
```
We only want to add fzf to zshrc if we don't have it.